server/src/main/java/org/elasticsearch/node/PluginComponentBinding.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.node;
+
+import org.elasticsearch.plugins.ExtensiblePlugin;
+import org.elasticsearch.plugins.Plugin;
+
+/**
+ *<p>
+ * Describes the relationship between an interface and an implementation for a {@link Plugin} component. All {@link Plugin}'s
+ * provide a collection of components via the createComponents(...) that are made available for
+ * dependency injection. An {@link ExtensiblePlugin} may want to expose an interface such that extensions can change the implementation.
+ * Usage of this class allows the interface to made available for dependency injection where the implementation is defined at runtime.
+ *</p>
+ *     Usage:
+ * <pre>{@code
+public class MyPlugin extends Plugin implements ExtensiblePlugin
+{
+    @Override
+    public Collection<Object> createComponents(...) {
+        List<Object> components = new ArrayList<>();
+        components.add(new PluginComponentInterface<>(MyInterface.class, this.myImplementation));
```

Comment:
This should be changed to PluginComponentBinding

Reply:
done [`1b3560b` (#94347)](https://github.com/elastic/elasticsearch/pull/94347/commits/1b3560b8e0f0f741d7c799c5840757baa2c2e6ea)